bob-apps/bin/bobt.rs

```diff
@@ -0,0 +1,434 @@
+        let size = self.get(key).await?;
+        log::debug!("Delete {} with size {}", key, size);
+        let req = self
+            .http_client
+            .delete(format!("{}/data/{}", self.settings.api_uri, self.settings.convert_key(key)))
+            .build()
+            .map_err(|e| e.to_string())?;
+        let sw = Stopwatch::new();
+        let _ = self
+            .http_client
+            .execute(req)
+            .await
+            .map_err(|e| e.to_string());
+        self.delete_time += sw.elapsed();
+        self.delete_count += 1;
+        let res = self.get(key).await;
+        match res {
+            Ok(new_size) => Err(format!("{} => {}", size, new_size)),
+            Err(_) => Ok(size),
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+struct Settings {
+    count: u64,
+    start_id: u64,
+    end_id: u64,
+    max_size: usize,
+    key_size: usize,
+    uri: Uri,
+    api_uri: Uri,
+}
+
+impl Settings {
+    fn new() -> Self {
+        let matches = get_matches();
+        Self {
+            count: Self::get_count(&matches),
+            uri: Self::get_uri(&matches),
+            start_id: Self::get_start_id(&matches),
+            end_id: Self::get_end_id(&matches),
+            max_size: Self::get_max_size(&matches),
+            key_size: Self::get_key_size(&matches),
+            api_uri: Self::get_api_uri(&matches),
+        }
+    }
+
+    fn get_count(matches: &ArgMatches) -> u64 {
+        matches
+            .value_of(COUNT_ARG_NAME)
+            .expect("required")
+            .parse()
+            .expect("should be u64")
+    }
+
+    fn get_max_size(matches: &ArgMatches) -> usize {
+        matches
+            .value_of(MAX_SIZE_ARG_NAME)
+            .expect("has default")
+            .parse()
+            .expect("should be usize")
+    }
+
+    fn get_start_id(matches: &ArgMatches) -> u64 {
+        matches
+            .value_of(START_ID_ARG_NAME)
+            .expect("has default")
+            .parse()
+            .expect("should be u64")
+    }
+
+    fn get_end_id(matches: &ArgMatches) -> u64 {
+        matches
+            .value_of(END_ID_ARG_NAME)
+            .expect("has default")
+            .parse()
+            .expect("should be u64")
+    }
+
+    fn get_uri(matches: &ArgMatches) -> Uri {
+        matches
+            .value_of(URI_ARG_NAME)
+            .expect("has default value")
+            .parse()
+            .expect("wrong format of url")
+    }
+
+    fn get_key_size(matches: &ArgMatches) -> usize {
+        matches
+            .value_of(KEY_SIZE_ARG_NAME)
+            .expect("has default value")
+            .parse()
+            .expect("wrong format of url")
+    }
+
+    fn get_api_uri(matches: &ArgMatches) -> Uri {
+        matches
+            .value_of(API_ADDRESS_ARG_NAME)
+            .expect("has default value")
+            .parse()
+            .expect("wrong format of url")
+    }
+
+    fn convert_key(&self, key: u64) -> String {
```

Comment:
Acutally, REST API should understand integer keys without problem, so you don't need to convert it into HEX representation